crates/aptos-profiler/src/cpu_profiler.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use crate::{CpuProfilerConfig, Profiler};
+use crate::utils::{convert_svg_to_string, create_file_with_parents};
+use std::{thread, time, path::PathBuf};
+
+pub struct CpuProfiler {
+    duration: u64,
+    frequency: i32,
+    cpu_profiling_data_file: PathBuf,
+}
+
+impl CpuProfiler {
+    pub(crate) fn new(config: &CpuProfilerConfig) -> Self {
+        Self {
+            duration: config.duration,
+            frequency: config.frequency,
+            cpu_profiling_data_file: config.cpu_profiling_result.clone(),
+        }
+    }
+}
+
+impl Profiler for CpuProfiler {
+    fn start_profiling(&self) -> Result<()> {
+        let guard = pprof::ProfilerGuard::new(self.frequency).unwrap();
+        thread::sleep(time::Duration::from_secs(self.duration));
+
+        if let Ok(report) = guard.report().build() {
+            let file = create_file_with_parents(self.cpu_profiling_data_file.as_path())?;
+
+            report.flamegraph(file).unwrap();
+        };
+        
+        Ok(())
+    }
+    // End profiling
+    fn end_profiling(&self) -> Result<()> {
```

Comment:
do we really need an end_profiling function?

Reply:
it seems that we will for executor benchmark integration

Reply:
How would that work? Are you going to stop the profiler by the duration_secs or manually?

Reply:
currently, it stops after duration_secs by default. the end profiling would stop it manually in concept. I currently do it for memory profiling, but it seems trickier for cpu because of the crate I'm using.

Reply:
and we'll likely need it for executor benchmark integration because when we are running it for the entire duration, it's hard to give an exact duration, so we will start with duration = max int and stop when needed. this is unimplemented though.

Reply:
In that case, I'd suggest to separate the interfaces, and do not store the duration as a member variable of the struct.

for profiling a predefined duration, take it as a parameter. e.g. fn profile_for(duration_secs: u64) -> Result<...>
for manually starting/stopping, have a fn start_profiling() -> Result<...> and fn stop_profiling() -> Result<...>

Reply:
The thing is, for CPU profiling, we have to provide a duration because it automatically stops when ProfilerGuard object goes out of scope. In that case, I think it would work if I give have max int as the duration for manually starting/stopping. Although it's not here yet, I'm currently working on adding a function to load configs from a yaml file, instead of just having default configs like here, so that will give the option for providing input, but I can create separate functions for running for a duration and running/stopping manually as you said as well.

Reply:
`I think it would work if I have max int as the duration for manually starting/stopping.`
OK

`so that will give the option for providing input`
Sounds good, however yaml file might be an overkill. http parameter for node and commandline flag for benchmark seem to be more clean and elegant.

`but I can create separate functions for running for a duration and running/stopping manually as you said as well.`
Yes, this is what I want, the interface need to be clean and well document.

Reply:
I've pushed an implementation. However, for CPU profiling without a duration, it starts the profiling, but stopping and generating results are unimplemented with the end_profiling function. Still exploring options on that as it wasn't a clear need before.